src/tests/ftest/pool/bad_evict.py

```diff
@@ -71,12 +71,12 @@ def test_evict(self):
                 pool.pool.uuid[4] = 244
 
             # evict the pool
-            pool.pool.evict()
+            self.get_dmg_command().pool_evict(pool.pool.get_uuid_str())
```

Comment:
```suggestion
            pool.evict()
```
Call pool.evict() if you want to use your new evict(). I guess you can use get_dmg_command(), but the original test is trying to do everything through the TestPool instance, so I think using your new evict() makes more sense. If you use your new evict(), I believe you need to set control_method: dmg under pool in the yaml.

Reply:
I just added self.pool.evict() for future use as I noticed it was missing but I think there need to be a lot of re-structuring of the test in order to use this method. So pool.evict() calls self.pool.uuid and if I use that, it will continue to give a continuous pass (even for null, and invalid uuids) and won't use the updated pool.uuid as it is inheriting original self.pool object.
But if we want to use invalid, null and actually self.pool.uuid we need to call self.get_dmg_command().pool_evict().
Having said that, I agree it is doable but will just need a whole lot of re-jig.